alpha/apps/kaltura/lib/kUploadTokenMgr.php

```diff
@@ -117,6 +117,24 @@ public function getUploadToken()
 	 */
 	public function uploadFileToToken($fileData, $resume = false, $resumeAt = -1)
 	{
+		if (!$this->_uploadToken->getFileName())
+		{
+			$this->_uploadToken->setFileName($fileData['name']);
+		}
+		$entry = $this->getAttachedEntry();
+		if ($entry && $entry->getType() == entryType::MEDIA_CLIP && !$entry->getMediaType())
```

Comment:
i think we can move all the logic back to the end of the function (the place where it was before this change)